src/name/ip_address.rs

```diff
@@ -317,7 +317,9 @@ pub(crate) fn parse_ipv4_address(ip_address_: &[u8]) -> Result<IpAddrRef, AddrPa
                     // No octet can be greater than 255.
                     return Err(AddrParseError);
                 }
-                octets[octet] = radix10_to_octet(&current_octet[..current_size]) as u8;
+                octets[octet] =
+                    TryInto::<u8>::try_into(radix10_to_octet(&current_octet[..current_size]))
+                        .expect("invalid character");
```

Comment:
I'm adding these `TryInto` due to clippy being executed with `-D clippy::as-conversions`. If there is a better/more idiomatic way to shape this code I'm happy adapting.

Reply:
Same comment applies to the rest of `TryInto` conversions in this commit.